quickwit/quickwit-search/src/leaf.rs

```diff
@@ -1226,6 +1263,7 @@ pub async fn leaf_search(
     let incremental_merge_collector = Arc::new(Mutex::new(incremental_merge_collector));
 
     for (split, mut request) in split_with_req {
+        timeout.err_on_timeout()?;
```

Comment:
I think if you timeout on the server side using a tower layer, it will cancel the future and do what you want without the TimeoutHandle.

Reply:
I tested the current behavior and the client side was unaffected from the server timeouts and kept running. There's no special logic in the tower timeout layer to forward the timeout to the client side, so not sure how that could be the case.

But there's another problem: If we dispatch locally, it doesn't use the grpc stuff, so we need to have some timeout for that case.

I replaced `TimeoutHandle` with a tokio timeout, that should be much nicer overall.